proggy.cpp

```diff
@@ -92,25 +133,33 @@ std::string toLower(std::string const & data) {
     return ret;
 }
 
-Customer const applyDiscount(Customer const & cust, std::string comp) {
- 
-   auto ret = cust;
+//Customer const applyDiscount(Customer const & cust, std::string comp) {
+//** KOOS
+//** Pass in by ref, then sends a copy back??
+void applyDiscount(Customer& cust, std::string comp) {
+
+//   auto ret = cust;
+//**KOOS: Dont create copies of objects.. apply discount to source
 
    auto lfname = toLower(cust.fname);
    if (lfname.compare(toLower(comp)) == 0){
        std::cout<<"apply discount!"<<std::endl;
-       ret.spent = cust.spent*0.5;
+//       ret.spent = cust.spent*0.5;
+//**KOOS: apply discount directly to source
+       cust.spent *= 0.5;
```

Comment:
hah! I had this in the first version and changed it when I got yelled at by the compiler